### ISSUE TITLE
Get more frequently updates.

### ISSUE THREAD

**Issue by teneko** [Mongo2Go/Mongo2Go#154]
Is there the possiblity to get more frequently the updates for MongoDB library related udpates? Huge part of my current tests all fail, because Mongo2Go "stucks" on MongoDB.* 3.0.0. 🤣

**Comment by JohannesHoppe** [comment]
Hello, hello! 🙂👋

Please read https://github.com/Mongo2Go/Mongo2Go/pull/153#issuecomment-2506442885 . It's important that I can reproduce a byte-perfect state of the same version. Then I can trust you, and you can trust me. In my opinion nobody should blindly execute my binaries, and therefore we always need two parties for an update of the binaries.

see #155

**Comment by teneko** [comment]
Mhh, maybe I do not undertand it well, but I do not talk about the MongoDB binaries per sé. I would like to distinct between MongoDB server binaries you may update via your MongoDownloader mentioned in #153, and C# MongoDB libraries you update via NuGet packages. This issue is about the latter.

Btw. I have written you an email. Thanks in advance.

**Comment by JohannesHoppe** [comment]
ah, ok, it thought you wanted to update the mongo daemon, which is very outdated, see https://github.com/Mongo2Go/Mongo2Go/tree/main/tools .

To update the MongoDB.Driver to version 3.1.0 you must also sent a PR. This is a minor version update.

**Comment by teneko** [comment]
Sorry the late answer, of course I will create another PR soon. :)

**Comment by teneko** [comment]
@JohannesHoppe PR is present. 🙂

**Comment by teneko** [comment]
If you want we can further discuss strategies to overcome the nuget package size problematic and the lack of automatic package version upgrades.

**Comment by teneko** [comment]
Any feedback please??

**Comment by JohannesHoppe** [comment]
Today I found the time to merge this and test it. Thank you for your patience. 🙂🙏
The new version is now available:

https://github.com/Mongo2Go/Mongo2Go/releases/tag/v4.1.0

**Comment by teneko** [comment]
Hey thanks for the answer. I really must admit, that I really like open-source but it has the bad habit to be short-living, short-handed. What I want to say is, that I would wish short communication paths, more interest to everything to produce more output of any kind. My patience is getting more and more limited which is the reason I deleted comments because it feels not right to participate in a process to discuss a process to the effective result stretched over many months. I do not want to blame anyone. I just collaborating via issue comments is not that fun at all.

**Comment by JohannesHoppe** [comment]
> Hey thanks for the answer. I really must admit, that I really like open-source but it has the bad habit to be short-living, short-handed. What I want to say is, that I would wish short communication paths, more interest to everything to produce more output of any kind. My patience is getting more and more limited which is the reason I deleted comments because it feels not right to participate in a process to discuss a process to the effective result stretched over many months. I do not want to blame anyone. I just collaborating via issue comments is not that fun at all.

Thanks for your honesty — I totally get your frustration.

But open source, at its core, means contributing code first. Issues and comments come second. If something is important to you, the fastest way forward is usually to open a PR. Discussing things for months without action doesn’t scale well, especially when maintainers (me, and only me) are working on this in their spare time.

That said, I really appreciate the engagement and your interest in improving things.

**Comment by teneko** [comment]
That's not true. By saying "the fastest way forward is usually to open a PR", you expect producing a result before even knowing the result is the one you and me accept and to be honest more you than me, because you have expectations, for example the MongoDB binaries must be available through NuGet because of your mentioned limitations in other contexts. So the reality is, at first we would have to find a compromise via communication before actually working on a PR. Then we would have the problem I tried to emphasive above: to deal with such a complex topic via issue comments is time consuming, takes several "sessions" may it be weeks or months, sounds not as much fun. As time goes, I also have to move on on topics I have to push forward to get my food on the table, so my time frame are counted in weeks rather than months.

That said I cannot blindly program something without having a plan and hope for the best - an accepted PR. And sadly the time frame I mentioned before is already closed and I cannot invest more time on this topic. I had to copy the project with respect to the license and updated the NuGet packages myself.

**Comment by JohannesHoppe** [comment]
> That's not true. By saying "the fastest way forward is usually to open a PR", you expect producing a result before even knowing the result is the one you and me accept and to be honest more you than me, because you have expectations, for example the MongoDB binaries must be available through NuGet because of your mentioned limitations in other contexts. So the reality is, at first we would have to find a compromise via communication before actually working on a PR. Then we would have the problem I tried to emphasive above: to deal with such a complex topic via issue comments is time consuming, takes several "sessions" may it be weeks or months, sounds not as much fun. As time goes, I also have to move on on topics I have to push forward to get my food on the table, so my time frame are counted in weeks rather than months.
> 
> That said I cannot blindly program something without having a plan and hope for the best - an accepted PR. And sadly the time frame I mentioned before is already closed and I cannot invest more time on this topic. I had to copy the project with respect to the license and updated the NuGet packages myself.

Forking is a valid and respected way to move forward in open source. If your fork has improvements or updated binaries, feel free to make it public. I’d be happy to take a look and see what can be merged back into Mongo2Go. 🙂👍
